src/App.tsx

```diff
@@ -27,23 +27,52 @@ const purchases: IPurchase[] = [
         date: '25.07.21',
         isEdit: false,
     },
-]
+];
 
 const App = (): JSX.Element => {
-    const [purchaseList, setPurchaseList] = useState<IPurchase[]>(purchases)
-    const totalPrice = purchaseList.reduce((total , item) => total + item.price, 0)
+    const [purchaseList, setPurchaseList] = useState<IPurchase[]>(purchases);
+    const totalPrice = purchaseList.reduce((total , item) => total + item.price, 0);
 
     const addPurchase = (purchase: IPurchase) => {
-        setPurchaseList([...purchaseList, purchase])
-    }
+        setPurchaseList([...purchaseList, purchase]);
+    };
+
+    const onPlaceChange = (place: string, purchase: IPurchase) => {
+        const purchaseIndex = purchaseList.findIndex((item) => item.id === purchase.id);
+        purchaseList[purchaseIndex].place = place;
+        setPurchaseList([...purchaseList]);
+    };
```

Comment:
findIndex might return -1, you don't check this condition with if.

Reply:
do not modify original array. Create new array with destructuring and work with it instead.